File: src/java/com/zenesis/qx/remote/annotations/PropertyDate.java
package com.zenesis.qx.remote.annotations;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Retention(RetentionPolicy.RUNTIME)
public @interface PropertyDate {

    public enum DateValues {
        DATE, DATE_TIME
    }
    public DateValues value() default DateValues.DATE_TIME;
    public boolean zeroTime() default true;
}

File: WebContent/qso-lib/source/class/com/zenesis/qx/remote/annotations/PropertyDate.js
qx.Class.define("com.zenesis.qx.remote.annotations.PropertyDate", {
  extend: qx.core.Object,

  properties: {
    value: {
      init: "dateTime",
      nullable: false,
      check: ["date", "dateTime"]
    },

    zeroTime: {
      init: false,
      nullable: false,
      check: "Boolean"
    }
  }
});
